Firestore/Swift/Source/PropertyWrapper/FirestoreQuery.swift

```diff
@@ -0,0 +1,176 @@
+  }
+
+  public static func whereField(_ lhs: String, isNotIn rhs: [Any]) -> FIRPredicate {
+    .isNotIn(lhs, rhs)
+  }
+
+  public static func whereField(_ lhs: String, arrayContains rhs: Any) -> FIRPredicate {
+    .arrayContains(lhs, rhs)
+  }
+
+  public static func whereField(_ lhs: String, arrayContainsAny rhs: [Any]) -> FIRPredicate {
+    .arrayContainsAny(lhs, rhs)
+  }
+
+  public static func whereField(_ lhs: String, isLessThan rhs: Any) -> FIRPredicate {
+    .isLessThan(lhs, rhs)
+  }
+
+  public static func whereField(_ lhs: String, isGreaterThan rhs: Any) -> FIRPredicate {
+    .isGreaterThan(lhs, rhs)
+  }
+
+  public static func whereField(_ lhs: String, isLessThanOrEqualTo rhs: Any) -> FIRPredicate {
+    .isLessThanOrEqualTo(lhs, rhs)
+  }
+
+  public static func whereField(_ lhs: String, isGreaterThanOrEqualTo rhs: Any) -> FIRPredicate {
+    .isGreaterThanOrEqualTo(lhs, rhs)
+  }
+
+  public static func order(by lhs: String, descending rhs: Bool = false) -> FIRPredicate {
+    .orderBy(lhs, rhs)
+  }
+
+  public static func limit(to lhs: Int) -> FIRPredicate {
+    .limitTo(lhs)
+  }
+}
+
+@available(iOS 13.0, *)
+internal class QueryStore<T: Decodable>: ObservableObject {
+  @Published var items: [T] = []
+
+  private let firestore = Firestore.firestore()
+  private var listener: ListenerRegistration? = nil
+
+  init(collectionPath: String, predicates: [FIRPredicate]) {
+    setupListener(from: collectionPath, withPredicates: predicates)
+  }
+
+  deinit {
+    removeListener()
+  }
+
+  private func setupListener(from collectionPath: String,
+                             withPredicates predicates: [FIRPredicate]) {
+    var query: Query = firestore.collection(collectionPath)
+
+    for predicate in predicates {
+      switch predicate {
+      case let .isEqualTo(lhs, rhs):
+        query = query.whereField(lhs, isEqualTo: rhs)
+      case let .isIn(lhs, rhs):
+        query = query.whereField(lhs, in: rhs)
+      case let .isNotIn(lhs, rhs):
+        query = query.whereField(lhs, notIn: rhs)
+      case let .arrayContains(lhs, rhs):
+        query = query.whereField(lhs, arrayContains: rhs)
+      case let .arrayContainsAny(lhs, rhs):
+        query = query.whereField(lhs, arrayContainsAny: rhs)
+      case let .isLessThan(lhs, rhs):
+        query = query.whereField(lhs, isLessThan: rhs)
+      case let .isGreaterThan(lhs, rhs):
+        query = query.whereField(lhs, isGreaterThan: rhs)
+      case let .isLessThanOrEqualTo(lhs, rhs):
+        query = query.whereField(lhs, isLessThanOrEqualTo: rhs)
+      case let .isGreaterThanOrEqualTo(lhs, rhs):
+        query = query.whereField(lhs, isGreaterThanOrEqualTo: rhs)
+      case let .orderBy(lhs, rhs):
+        query = query.order(by: lhs, descending: rhs)
+      case let .limitTo(lhs):
+        query = query.limit(to: lhs)
+      }
+    }
+
+    listener = query.addSnapshotListener { [weak self] snapshot, error in
+      if let error = error {
+        print(error)
```

Comment:
We still need to decide how to deal with errors.

Reply:
Absolutely right. Currently I set the items array to empty and print out errors.

Reply:
Would it be too strange to have  projectedValue with the following signature:
```
public var projectedValue: AnyPublisher<[Result<T, Error>], Never>
```
?
That is: If you're interested in getting errors, then you can get a publisher of an array of results.
That signature of course doesn't match top level errors - I don't know if it would be too hacky if a top level error was represented as an array containing a single failure result?

Reply:
Why would you use a Result type and set the error type to Never? Wouldn't it be much easier to use
`public var projectedValue: AnyPublisher<[T], FirebaseError>`
?

Reply:
Perhaps adding the tests will help clarify an error handling strategy?